src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -403,6 +411,39 @@ bool SettingsLoader::DisableDeletedProfiles()
     return newGeneratedProfiles;
 }
 
+bool winrt::Microsoft::Terminal::Settings::Model::implementation::SettingsLoader::RemapColorSchemeForProfile(const winrt::com_ptr<winrt::Microsoft::Terminal::Settings::Model::implementation::Profile>& profile)
+{
+    bool modified{ false };
+    auto remapForAppearance = [&](const IAppearanceConfig& appearance) {
+        if (auto schemeName{ appearance.LightColorSchemeName() }; !schemeName.empty())
+        {
+            if (auto found{ userSettings.colorSchemeRemappings.find(schemeName) }; found != userSettings.colorSchemeRemappings.end())
+            {
+                appearance.LightColorSchemeName(found->second);
+                modified = true;
+            }
+        }
+
+        if (auto schemeName{ appearance.DarkColorSchemeName() }; !schemeName.empty())
+        {
+            if (auto found{ userSettings.colorSchemeRemappings.find(schemeName) }; found != userSettings.colorSchemeRemappings.end())
+            {
+                appearance.DarkColorSchemeName(found->second);
+                modified = true;
+            }
+        }
+    };
+
+    remapForAppearance(profile->DefaultAppearance());
+
+    if (auto unfocusedAppearance{ profile->UnfocusedAppearance() })
+    {
+        remapForAppearance(unfocusedAppearance);
+    }

```

Comment:
An alternative way to express this would be something like this:
```cpp
const IAppearanceConfig appearances[] = {
    profile->DefaultAppearance(),
    profile->UnfocusedAppearance(),
};

for (const auto& appearance : appearances) {
    if (!appearance) {
        continue;
    }

    // the lambda contents
}
```

Reply:
I prefer this, thanks